scripts/irods/test/test_federation.py

```diff
@@ -1442,3 +1471,90 @@ def test_icp_tree_remotezone_in_home(self):
+        for key, val in test.settings.FEDERATION.__dict__.items():
+            if not key.startswith('__'):
+                self.config[key.lower()] = val
+
+        self.config['local_zone'] = self.user_sessions[0].zone_name
+        if test.settings.FEDERATION.REMOTE_IRODS_VERSION < (4, 0, 0):
+            test.settings.FEDERATION.REMOTE_VAULT = '/home/irods/irods-legacy/iRODS/Vault'
+
+        self.admin = session.make_session_for_existing_admin()
+
+    def tearDown(self):
+        self.admin.__exit__()
+        super(test_dynamic_peps, self).tearDown()
+
+    def test_peps_for_parallel_mode_transfers__issue_5017(self):
+        remote_home_collection = self.admin.remote_home_collection(test.settings.FEDERATION.REMOTE_ZONE)
+        filename = 'test_peps_for_parallel_mode_transfers__issue_5017'
+        local_file = os.path.join(self.admin.local_session_dir, filename)
+        logical_path = os.path.join(remote_home_collection, filename)
+        metadata_resource = 'test_peps_for_parallel_mode_transfers_resource'
+        file_size = 40 * 1024 * 1024 # 40MB
+        peps_to_check = ['data-obj-put-pre', 'data-obj-put-except', 'data-obj-put-finally']
+        attr = 'test_peps_for_parallel_mode_transfers__4404_put'
+
+        try:
+            if not os.path.exists(local_file):
+                lib.make_file(local_file, file_size)
+
+            # PEPs will fire locally on connected server, so metadata will applied to local resource
+            parameters = {}
+            parameters['resource'] = metadata_resource
+            put_peps = rule_texts[self.plugin_name][self.class_name][attr].format(**parameters)
+            print(put_peps)
+
+            def reset_resource(self):
+                self.admin.run_icommand(['iadmin', 'rmresc', metadata_resource])
+                self.admin.assert_icommand(['iadmin', 'rum'])
+                lib.create_passthru_resource(metadata_resource, self.admin)
+
+            with temporary_core_file() as core:
+                time.sleep(1)  # remove once file hash fix is committed #2279
+                core.add_rule(put_peps)
+                time.sleep(1)  # remove once file hash fix is committed #2279
+
+                reset_resource(self)
+
+                # put a new data object and ensure success
+                out, err, ec = self.admin.run_icommand(['iput', local_file, logical_path])
+                self.assertEqual(0, ec)
+
+                for pep in ['data-obj-put-pre', 'data-obj-put-post', 'data-obj-put-finally']:
+                    lib.delayAssert(
+                        lambda: lib.metadata_attr_with_value_exists(self.admin, attr, pep),
+                        interval=1,
+                        maxrep=10
+                    )
+
+                self.assertFalse(lib.metadata_attr_with_value_exists(self.admin, attr, 'pep-obj-put-except'))
+
+                reset_resource(self)
+
+                # put to same logical path, targeting a non-existent replica (resulting in error)
```

Comment:
Change to: put to same logical path without force flag, resulting in an error and (hopefully) triggering the except pep